### DISCUSSION TITLE
Deploying a CRA served by a express server

### DISCUSSION THREAD

**Discussion by andresz1** [vercel/vercel#4336] [General] [answered]
Hi! I'm performing a small PoC of how [now](https://github.com/zeit/now) could be used to deploy a custom Server-Side Rendering React Application. I wanted to start with something really simple so I decided to create a brand new app using [CRA](https://create-react-app.dev/docs/getting-started/) and an [express](https://expressjs.com/) server to serve the `dist` files (this is my first time using custom steps).

**server/index.js**
```js
const port = 4000
const express = require('express')
const compression = require('compression')
const basicAuth = require('express-basic-auth')

const app = express()

app.use(compression())
app.use(express.static('./'))

app.get('*', (req, res) => res.sendFile(('/index.html')));

app.listen(port, '0.0.0.0', () => {
    console.log('listen');
});
```

What I want to do is: `yarn install`, `yarn build` and then `node server/index.js`. I'm little bit confused of how I should do that. Here is my `now.json` file. It uses `@now/static-build` to build the app and `@now/node-server` to run the server. Nevertheless, I always get (I've tried changing the `static` folder to several options):
```
ERROR	Error: ENOENT: no such file or directory, stat '/index.html'
```
It's like those file couldn't be access from the serverless function even though  I can see them in the `source` output

![Screenshot 2020-05-11 at 18 19 16](https://user-images.githubusercontent.com/6877967/81585127-feb62380-93b3-11ea-8645-82bf257765d8.png)

**now.json**
```js
{
    "version": 2,
    "public": false,
    "builds": [
      { "src": "package.json", "use": "@now/static-build", "config": { "distDir": "build" } },
      { "src": "/server/index.js", "use": "@now/node-server" }
    ],
    "routes": [
      { "src": "/.*", "dest": "/server/index.js" }
    ]
  }
```

Thank you so much in advance. I'd highly appreciate any help or suggestion.

**Comment by styfle** [comment, marked as answer]
You don't need express to serve static files.

You can rename `/static` to `/public` or add a `build` script to your `package.json` file that generates files in the `public` directory.

Also, rename `/server/index.js` to `/api/index.js` and then you can remove `now.json` completely.

Read more about [Build Step](https://vercel.com/docs/v2/build-step) and [Serverless Functions](https://vercel.com/docs/v2/serverless-functions/introduction) in the docs.
